### ISSUE TITLE
Please add auto reconnect feature

### ISSUE THREAD

**Issue by guglez** [airdcpp-web/airdcpp-webclient#210]
Hi,

My client can't connect to a password-protected hub sometimes (when hub is full for example or for any other reason). It will be great if client will be able to reconnect automatically.

Thanks,
guglez

**Comment by maksis** [comment]
The client should reconnect automatically after a few minutes. Is it an ADC hub and what is the hubsoft?

**Comment by guglez** [comment]
It's a 0.9.8e-r2 VerliHub. I will check if client will reconnect automatically.

**Comment by guglez** [comment]
I can confirm that client do not reconnect automatically during startup. Also hub is still asking for a password (I set the password in favorites menu).

I have only one hub in favorites and than hub uses password to connect.

**Comment by maksis** [comment]
Could you post the information that is requested when submitting new bug reports (as this is not a feature request)?

Does it attempt to reconnect but it stops because of the password prompt?

**Comment by guglez** [comment]
### Current UI and client versions:

Application version
AirDC++w 1.4.1 x86_64
Web UI version
1.4.0
Web UI build date
December 9, 2016 11:12 AM
Started
5 minutes ago
Active sessions
1
Server threads
4



### Operating system:
Ubuntu 14.04.5 LTS
Linux player-pc 3.19.0-78-generic #86~14.04.1-Ubuntu SMP Tue Dec 6 17:58:35 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux


### Steps to reproduce the issue:

Add manually hub to favorites, set password, enable autoconnect.
Restart application. It will not even try to connect (I checked that via tcpdump -i eth0 port 4111 - was unable to see any packets).

Known workaround: connect to hub manually then add the hub to favorites via dropdown menu.



##########
Bottom line. Looks like that I just found a workaround. The hub I'm using has several dns names. I used the sort one at the beginning.
Let's say the short name is dc.lan
but the full name which is visible via web interface is dchub://hub.dc.lan:4111

Now it works as expected.

**Comment by maksis** [comment]
The address of the favorite hub entry must match the hub address that you are trying to connect to. Are you saying that you were connected to a different address that you had added in favorites (and how did that happen)? Did the hub that you had in favorites redirect you to a different address? The address shouldn't be altered otherwise.

**Comment by guglez** [comment]
HUB administrators can create as many DNS names for the host as they wish. In my case I have several different names which will point you to the same IP address.


Initially I added short address (dc.lan) to favorites by hand. But later I just manually connected to that address (dc.lan). And added it to favorites via dropdown menu. And now in favorites I can see dchub://hub.dc.lan:4111

I need to say that eiskaltdcpp has exactly the same behavior.

**Comment by maksis** [comment]
Yes, that may happen since address is generally the only identifier for a hub
